doc/docs/usage/mtls-setup.md

```diff
@@ -0,0 +1,208 @@
+CN = ank-agent
+
+[v3_req]
+subjectAltName = @alt_names
+extendedKeyUsage = clientAuth
+
+[alt_names]
+# This certificate can only be used for agents with the names 'agent_A' or 'agent_B'
+# To allow the usage for any agent use the character '*'
+# like: DNS.1 = *
+DNS.1 = agent_A
+DNS.2 = agent_B
+
+```
+
+Generate ank-agent key:
+
+```shell
+sudo openssl genpkey -algorithm ED25519 -out "/etc/ankaios/certs/ank-agent-key.pem"
+```
+
+Generate ank-agent certificate signing request:
+
+```shell
+sudo openssl req -config "./ank-agent.cnf" -new -key "/etc/ankaios/certs/ank-agent-key.pem" -out "./ank-agent.csr"
+```
+
+Generate ank-agent certificate:
+
+```shell
+sudo openssl x509 -req -in "./ank-agent.csr" -CA "/etc/ankaios/certs/ca.pem" -CAkey "/etc/ankaios/certs/ca-key.pem" -extensions v3_req -extfile "./ank-agent.cnf" -out "/etc/ankaios/certs/ank-agent.pem"
+```
+
+## Generate key and certificate for the CLI `ank`
+
+Construct an [OpenSSL configuration file](https://www.openssl.org/docs/manmaster/man5/config.html) named `ank.cnf`. You are welcome to include additional fields if necessary:
+
+```ini title="ank.cnf"
+[req]
+distinguished_name = req_distinguished_name
+req_extensions = v3_req
+prompt = no
+[req_distinguished_name]
+CN = ank
+
+[v3_req]
+subjectAltName = @alt_names
+extendedKeyUsage = clientAuth
+
+[alt_names]
+DNS.1 = ank
+
+```
+
+Generate ank key:
+
+```shell
+sudo openssl genpkey -algorithm ED25519 -out "${XDG_CONFIG_HOME:-$HOME/.config}/ankaios/ank-key.pem"
+```
+
+Generate ank certificate signing request:
+
+```shell
+sudo openssl req -config "./ank.cnf" -new -key "${XDG_CONFIG_HOME:-$HOME/.config}/ankaios/ank-key.pem" -out "./ank.csr"
+```
+
+Generate ank certificate:
+
+```shell
+sudo openssl x509 -req -in "./ank.csr" -CA "/etc/ankaios/certs/ca.pem" -CAkey "/etc/ankaios/certs/ca-key.pem" -extensions v3_req -extfile "./ank.cnf" -out "${XDG_CONFIG_HOME:-$HOME/.config}/ankaios/ank.pem"
+```
+
+## Perform Ankaios installation with mTLS support
+
+To set up Ankaios with mutual TLS (mTLS) support, you need to supply the necessary mTLS certificates to the `ank-server`, `ank-agent`, and `ank` CLI components. Here's a step-by-step guide:
+
+### Install the `ank-server` with mTLS certificates
+
+```shell
+curl -sfL https://github.com/eclipse-ankaios/ankaios/releases/latest/download/install.sh | bash -s -- -s "--startup-config /etc/ankaios/state.yaml --ca_pem /etc/ankaios/certs/ca.pem --crt_pem /etc/ankaios/certs/ank-server.pem --key_pem /etc/ankaios/certs/ank-server-key.pem" -a "--name agent_A --ca_pem /etc/ankaios/certs/ca.pem --crt_pem /etc/ankaios/certs/ank-agent.pem --key_pem /etc/ankaios/certs/ank-agent-key.pem"
+```
+
+### Install the `ank-agent` with mTLS certificates
+
+```shell
+curl -sfL https://github.com/eclipse-ankaios/ankaios/releases/latest/download/install.sh | bash -s -- -s "--ca_pem /etc/ankaios/certs/ca.pem --crt_pem /etc/ankaios/certs/ank-agent.pem --key_pem /etc/ankaios/certs/ank-agent-key.pem"
```

Comment:
This puzzles me a bit. I needed only the previous step `Install the ank-server with mTLS support` to install all binaries `ank-server`, `ank-agent` and `ank`. This description gives user feeling, that it is necessary to do both steps, which is not true. This commad (to install ank-agent) broke something, so the agent was not running.

Reply:
Agree. See also my comment https://github.com/eclipse-ankaios/ankaios/pull/291#discussion_r1662359380